cpp/include/cuspatial/experimental/detail/linestring_intersection_with_duplicates.cuh

```diff
@@ -123,6 +194,63 @@ struct linestring_intersection_intermediates {
                            thrust::next(offsets->begin()));
   }
 
+  /// Given a flag array, remove the ith geometry if `flags[i] == 1`.
+  /// @pre flag array must have the same size as the geometry array.
+  template <typename FlagRange>
+  void remove_if(FlagRange flags, rmm::cuda_stream_view stream)
```

Comment:
I'd love to see more detailed comments about the steps in this function.

Reply:
Done in #851